docs/3.0/develop/transactions.mdx

```diff
@@ -126,7 +126,176 @@ def confirmation(transaction):
 ```
 </Tip>
 
-### Access data within transactions
+## Idempotency
+
+You can ensure sections of code are idempotent by wrapping them in a transaction. By specifying a `key` for your transaction, you can ensure that 
+your code is executed only once.
+
+For example, here's a flow that downloads some data from an API and writes it to a file:
+
+```python
+from prefect import task, flow
+from prefect.transactions import transaction
+
+
+@task
+def download_data():
+    """Imagine this downloads some data from an API"""
+    return "some data"
+
+
+@task
+def write_data(data: str):
+    """This writes the data to a file"""
+    with open("data.txt", "w") as f:
+        f.write(data)
+
+
+@flow(log_prints=True)
+def pipeline():
+    with transaction(key="download-and-write-data", write_on_commit=True) as txn:
```

Comment:
for my own understanding, i'm surprised `write_on_commit` is necessary given that it defaults to `True` on the `Transaction` object and providing a `key` hydrates a `store` under the hood. Is it actually necessary, and if so, _should_ it be necessary on the context manager?

Reply:
`write_on_commit=True` is necessary right now to write records when the transaction commits because the context manager uses the value from `should_persist_result()` by default, which returns `False` outside of a task or flow unless `PREFECT_RESULTS_PERSIST_BY_DEFAULT` is enabled.

I was surprised that I needed to specify `write_on_commit` explicitly, so I lean towards having it default to `True`.